// meta/HelloWorldService_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var HelloWorldService_pb = require('./HelloWorldService_pb.js');

function serialize_grpctest_HelloRequest(arg) {
  if (!(arg instanceof HelloWorldService_pb.HelloRequest)) {
    throw new Error('Expected argument of type grpctest.HelloRequest');
  }
  return new Buffer(arg.serializeBinary());
}

function deserialize_grpctest_HelloRequest(buffer_arg) {
  return HelloWorldService_pb.HelloRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_grpctest_HelloResponse(arg) {
  if (!(arg instanceof HelloWorldService_pb.HelloResponse)) {
    throw new Error('Expected argument of type grpctest.HelloResponse');
  }
  return new Buffer(arg.serializeBinary());
}

function deserialize_grpctest_HelloResponse(buffer_arg) {
  return HelloWorldService_pb.HelloResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// 服务HelloWorld  
var HelloWorldServiceService = exports.HelloWorldServiceService = {
  hello: {
    path: '/grpctest.HelloWorldService/hello',
    requestStream: false,
    responseStream: false,
    requestType: HelloWorldService_pb.HelloRequest,
    responseType: HelloWorldService_pb.HelloResponse,
    requestSerialize: serialize_grpctest_HelloRequest,
    requestDeserialize: deserialize_grpctest_HelloRequest,
    responseSerialize: serialize_grpctest_HelloResponse,
    responseDeserialize: deserialize_grpctest_HelloResponse,
  },
};

exports.HelloWorldServiceClient = grpc.makeGenericClientConstructor(HelloWorldServiceService);
